wpilibj/src/athena/java/edu/wpi/first/wpilibj/PeriodicRobot.java

```diff
@@ -0,0 +1,65 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) FIRST 2017. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package edu.wpi.first.wpilibj;
+
+import edu.wpi.first.wpilibj.hal.FRCNetComm.tInstances;
+import edu.wpi.first.wpilibj.hal.FRCNetComm.tResourceType;
+import edu.wpi.first.wpilibj.hal.HAL;
+import edu.wpi.first.wpilibj.livewindow.LiveWindow;
+
+/**
+ * PeriodicRobot implements the IterativeRobotBase robot program framework.
+ *
+ * <p>The PeriodicRobot class is intended to be subclassed by a user creating a robot program.
+ *
+ * <p>periodic() functions from the base class are called on an interval by a Notifier instance.
+ */
+public class PeriodicRobot extends IterativeRobotBase {
+  private static final double DEFAULT_PERIOD = 0.01;
+
+  private Notifier m_loop = new Notifier(() -> {
+    loopFunc();
+  });
+
+  /**
+   * Provide an alternate "main loop" via startCompetition().
+   */
+  public void startCompetition() {
+    HAL.report(tResourceType.kResourceType_Framework,
+                                   tInstances.kFramework_Periodic);
+
+    robotInit();
+
+    // Tell the DS that the robot is ready to be enabled
+    HAL.observeUserProgramStarting();
+
+    // loop forever, calling the appropriate mode-dependent function
+    LiveWindow.setEnabled(false);
+    m_loop.startPeriodic(DEFAULT_PERIOD);
+    while (true) {
+      try {
+        Thread.sleep(1000 * 60 * 60 * 24);
+      } catch (InterruptedException ex) {
+        continue;
```

Comment:
You actually want to handle it by passing the interrupt along with `Thread.currentThread().interrupt();`